File: src/main/java/dev/robothanzo/werewolf/model/SpeechSession.java
package dev.robothanzo.werewolf.model;

import dev.robothanzo.werewolf.database.documents.Session;
import lombok.Builder;
import lombok.Data;
import org.jetbrains.annotations.Nullable;

import java.util.LinkedList;
import java.util.List;

@Data
@Builder
public class SpeechSession {
    private long guildId;
    private long channelId;
    private Session session;
    @Builder.Default
    private List<Long> interruptVotes = new LinkedList<>();
    @Builder.Default
    private List<Session.Player> order = new LinkedList<>();
    @Nullable
    private Thread speakingThread;
    @Nullable
    private Long lastSpeaker;
    @Nullable
    private Runnable finishedCallback;
    @Builder.Default
    private long currentSpeechEndTime = 0;
    @Builder.Default
    private int totalSpeechTime = 0;
}

File: src/main/kotlin/dev/robothanzo/werewolf/model/SpeechSession.kt
package dev.robothanzo.werewolf.model

import dev.robothanzo.werewolf.database.documents.Session
import java.util.*

data class SpeechSession(
    val guildId: Long,
    val channelId: Long,
    val session: Session,
    val interruptVotes: MutableList<Long> = LinkedList(),
    val order: MutableList<Session.Player> = LinkedList(),
    var speakingThread: Thread? = null,
    var lastSpeaker: Long? = null,
    var finishedCallback: (() -> Unit)? = null,
    var currentSpeechEndTime: Long = 0,
    var totalSpeechTime: Int = 0,
    var shouldStopCurrentSpeaker: Boolean = false
)
